projects/dsp-ui/src/lib/viewer/values/time-value/time-value.component.html

```diff
@@ -7,6 +7,13 @@
   <span [formGroup]="form" class="parent-component-wrapper">
       <mat-form-field class="large-field child-value-component" floatLabel="never">
         <dsp-time-input #timeInput [formControlName]="'timeValue'" class="value" [errorStateMatcher]="matcher"></dsp-time-input>
+          <mat-error *ngIf="form.controls.timeValue.hasError('valueNotChanged') &&
+                (form.controls.timeValue.touched || form.controls.timeValue.dirty)">
+            <span class="custom-error-message">New value must be different than the current value.</span>
+          </mat-error>
+          <mat-error *ngIf="form.controls.timeValue.hasError('pattern')">
+            <span class="custom-error-message">New value must be in a valid time precision.</span>
```

Comment:
I'm having a hard time triggering this error, can you tell me what I need to do to trigger it and which browser?
Also, we could probably make this clearer by saying "A time value must follow the format HH:mm e.g. 16:30".

Reply:
> I'm having a hard time triggering the pattern error, can you tell me what I need to do to trigger it and which browser?
> Also, we could probably make this clearer by saying "A time value must follow the format HH:mm e.g. 16:30".

to trigger it, you would need to delete the minute precision. Regarding the error message, I can make it clearer as you suggest.

Reply:
> I'm having a hard time triggering the pattern error, can you tell me what I need to do to trigger it and which browser?
> Also, we could probably make this clearer by saying "A time value must follow the format HH:mm e.g. 16:30".

I just noticed that this check was superfluous as the material does not allow the user to enter any invalid pattern. I made the error message for the case of the missing hour of minute precisions clearer.

Reply:
Unfortunately the check is not superfluous in Safari since what forces the format is the "type" attribute and Safari doesn't support time as a type :(

However, since we have now upgraded to Angular 9 I can use [this library](https://github.com/danielmoncada/date-time-picker) which I wanted to use when I was first making the component to supply the user with a better way to enter a time. This will of course be a refactor and in a different PR.

Reply:
So, then I trust you will add the pattern check after you update the library.

Reply:
Yeah, not sure when or if the change will happen but I will make sure to handle the pattern check.

Reply:
> Unfortunately the check is not superfluous in Safari since what forces the format is the "type" attribute and Safari doesn't support time as a type :(
> 
> However, since we have now upgraded to Angular 9 I can use [this library](https://github.com/danielmoncada/date-time-picker) which I wanted to use when I was first making the component to supply the user with a better way to enter a time. It'll also homogenize the user experience independent of the browser they are using. This will of course be a refactor and in a different PR.

Good point, I had not tried with safari. I just added the pattern check to the time-input component. In Safari, you can trigger it now with replacing `:` with `;`.

Reply:
> > Unfortunately the check is not superfluous in Safari since what forces the format is the "type" attribute and Safari doesn't support time as a type :(
> > However, since we have now upgraded to Angular 9 I can use [this library](https://github.com/danielmoncada/date-time-picker) which I wanted to use when I was first making the component to supply the user with a better way to enter a time. It'll also homogenize the user experience independent of the browser they are using. This will of course be a refactor and in a different PR.
> 
> Good point, I had not tried with safari. I just added the pattern check to the time-input component. In Safari, you can trigger it now with replacing `:` with `;`.

Looks good to me now! @flavens Do you need another look before approving?

Reply:
@SepidehAlassi @mdelez no LGTM!